dhis-2/dhis-services/dhis-service-analytics/src/test/java/org/hisp/dhis/analytics/data/JdbcSubexpressionQueryGeneratorTest.java

```diff
@@ -167,6 +167,76 @@ ORGUNIT_DIM_ID, DimensionType.ORGANISATION_UNIT, getList(ouA)))
     assertEquals(expected, actual);
   }
 
+  @Test
+  void testGetSql_withPeriodOffset() {
+    OrganisationUnit ouA = createOrganisationUnit('A');
+
+    Period peA = createPeriod("202305");
+
+    QueryModifiers queryModsMin = QueryModifiers.builder().periodOffset(-1).build();
+
+    DataElement deA = createDataElement('A');
+    DataElement deB = createDataElement('A');
+
+    deB.setQueryMods(queryModsMin);
+
+    List<DimensionalItemObject> items = List.of(deA, deB);
+
+    String subexSql =
+        " case when coalesce(\"%s\",0) - coalesce(\"%s_minus_1\",0) > 0 then 1 else 0 end"
+            .formatted(deA.getUid(), deA.getUid());
+
+    SubexpressionDimensionItem subex = new SubexpressionDimensionItem(subexSql, items, null);
+
+    DataQueryParams params =
+        DataQueryParams.newBuilder()
+            .withDataType(DataType.NUMERIC)
+            .withTableName("analytics")
+            .withAggregationType(AnalyticsAggregationType.SUM)
+            .addDimension(
+                new BaseDimensionalObject(DATA_X_DIM_ID, DimensionType.DATA_X, getList(subex)))
+            .addFilters(
+                List.of(
+                    new BaseDimensionalObject(
+                        ORGUNIT_DIM_ID, DimensionType.ORGANISATION_UNIT, getList(ouA)),
+                    new BaseDimensionalObject(PERIOD_DIM_ID, DimensionType.PERIOD, getList(peA))))
+            .withPeriodType("monthly")
+            .build();
+
+    JdbcSubexpressionQueryGenerator target =
+        new JdbcSubexpressionQueryGenerator(manager, params, DATA_VALUE);
+
+    String expected =
+        "select "
```

Comment:
I suggest we use Java 15 text blocks here, with """, for better readability.

Reply:
Right. Though by using `\` at the end of each line, like we do many places in the analytics table code, the linebreak will be escaped, i.e. removed, and you will get it on one line.